src/main/java/org/opensearch/ad/AnomalyDetectorExtension.java

```diff
@@ -39,7 +44,14 @@ public ExtensionSettings getExtensionSettings() {
 
     @Override
     public List<ExtensionRestHandler> getExtensionRestHandlers() {
-        return List.of(new RestCreateDetectorAction());
+        List<ExtensionRestHandler> handler = null;
+        // Get the settings anc clusterService from createComponent
+        try {
+            handler = List.of(new RestCreateDetectorAction(null, null));
+        } catch (IOException e) {
```

Comment:
does instantiating this object throw an exception?

Reply:
Removed the exception from SDK: https://github.com/opensearch-project/opensearch-sdk-java/pull/234